src/mon/OSDMonitor.cc

```diff
@@ -341,10 +341,13 @@ bool is_unmanaged_snap_op_permitted(CephContext* cct,
 
 } // anonymous namespace
 
-void LastEpochClean::Lec::report(ps_t ps, epoch_t last_epoch_clean)
+void LastEpochClean::Lec::report(unsigned pg_num, ps_t ps,
+				 epoch_t last_epoch_clean)
 {
-  if (epoch_by_pg.size() <= ps) {
-    epoch_by_pg.resize(ps + 1, 0);
+  epoch_by_pg.resize(pg_num, 0);
+  if (ps >= pg_num) {
+    // removed PG
+    return;
```

Comment:
nit, just for readability. could you move `epoch_by_pg.resize(pg_num, 0);` down after the `if` block?

Reply:
ok done